js/store.js

```diff
@@ -29,31 +44,39 @@ const store = {
     return null;
   },
 
-  setFirstParty(param) {
-    /**
-     * @param object with hostname as key
-     * @todo validate the param.-
-     * @todo code to be updated in next PR
-     */
-    return this.set(param);
+  setFirstParty(hostname, data) {
+    if (!hostname) {
+      throw new Error('setFirstParty requires a valid hostname argument');
+    }
+
+    const websites = clone(this._websites);
+    websites[hostname] = data;
+    this._write(websites);
   },
 
-  async set(websites) {
-    /**
-      * @todo code to be updated in next PR
-    */
-    return await browser.storage.local.set({ websites });
+  setThirdParty(origin, target, data) {
```

Comment:
To piggy back on my first comment about using `document` instead of `hostname`, I believe here we are interested in the `document` URL, not `origin` URL. In extent, `setThirdParty(document, target, data)...`. I just created an issue to clarify the differences between `document`, `origin` and `target` URLs.